### ISSUE TITLE
Indentation bug on 2.0.0a1 release

### ISSUE THREAD

**Issue by alanmcruickshank** [sqlfluff/sqlfluff#4215]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

As raised on #4213 . This query is improperly fixed:

```sql
{% if False %}
SELECT 1
{% else %}
SELECT c
FROM t
WHERE c < 0
{% endif %}
```

on current main it fails as

```sql
{% if False %}
SELECT 1
{% else %}
SELECT c
    FROM t
    WHERE c < 0
{% endif %}
```

### Expected Behaviour

expected fix is that all three incorrectly indented lines should be fixed.

### Observed Behaviour

the `SELECT` line is not fixed.

### How to reproduce

see description above

### Dialect

ansi

### Version

ce5c5f228d92df339ef448e66dd3ef8ad859a9e2

### Configuration

default config

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Interestingly, when running `sqlfluff lint` - things work as expected - the first issue however has no fix proposed.

**Comment by alanmcruickshank** [comment]
Ok - fix is being rejected because:

 ```
INFO             * Discarding fixes that span multiple template blocks: [<LintFix: replace newline@[L:  3, P: 11] edt:'\n'->'\n    '>]
```

I don't think it *should* be rejected however, it's an edit fix on a literal newline. This is probably an off-by-one error in the rejection logic.
